wsd/wopi/WopiStorage.cpp

```diff
@@ -624,18 +624,13 @@ void WopiStorage::uploadLocalFileToStorageAsync(const Authorization& auth, LockC
 
     // TODO: Check if this URI has write permission (canWrite = true)
 
-    // Always invoke the callback with the result of the async upload.
-    ScopedInvokeAsyncRequestCallback<AsyncUploadCallback, StorageBase::AsyncUpload>
-        scopedInvokeCallback(
-            asyncUploadCallback,
-            StorageBase::AsyncUpload(
-                StorageBase::AsyncUpload::State::Error,
-                StorageBase::UploadResult(StorageBase::UploadResult::Result::FAILED)));
-
     //TODO: replace with state machine.
     if (_uploadHttpSession)
     {
         LOG_WRN("Upload is already in progress.");
+        asyncUploadCallback(StorageBase::AsyncUpload(
+            StorageBase::AsyncUpload::State::Error,
+            StorageBase::UploadResult(StorageBase::UploadResult::Result::FAILED)));
```

Comment:
maybe add "Already in progress." message as second param so it will be more detailed